cmd/abapEnvironmentCreateTag.go

```diff
@@ -0,0 +1,246 @@
+)
+
+func abapEnvironmentCreateTag(config abapEnvironmentCreateTagOptions, telemetryData *telemetry.CustomData) {
+
+	c := command.Command{}
+
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	var autils = abaputils.AbapUtils{
+		Exec: &c,
+	}
+
+	client := piperhttp.Client{}
+
+	err := runAbapEnvironmentCreateTag(&config, telemetryData, &autils, &client)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runAbapEnvironmentCreateTag(config *abapEnvironmentCreateTagOptions, telemetryData *telemetry.CustomData, com abaputils.Communication, client piperhttp.Sender) error {
+
+	connectionDetails, errorGetInfo := com.GetAbapCommunicationArrangementInfo(convertTagConfig(config), "")
+	if errorGetInfo != nil {
+		return errors.Wrap(errorGetInfo, "Parameters for the ABAP Connection not available")
+	}
+
+	// Configuring the HTTP Client and CookieJar
+	cookieJar, errorCookieJar := cookiejar.New(nil)
+	if errorCookieJar != nil {
+		return errors.Wrap(errorCookieJar, "Could not create a Cookie Jar")
+	}
+
+	client.SetOptions(piperhttp.ClientOptions{
+		MaxRequestDuration: 180 * time.Second,
+		CookieJar:          cookieJar,
+		Username:           connectionDetails.User,
+		Password:           connectionDetails.Password,
+	})
+
+	backlog, errorPrepare := prepareBacklog(config)
+	if errorPrepare != nil {
+		return fmt.Errorf("Something failed during the tag creation: %w", errorPrepare)
+	}
+
+	err := createTags(backlog, telemetryData, connectionDetails, client, com)
+
+	return err
+}
+
+func createTags(backlog []CreateTagBacklog, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	connection := con
+	connection.XCsrfToken = "fetch"
+	connection.URL = con.URL + "/sap/opu/odata/sap/MANAGE_GIT_REPOSITORY/Tags"
+	resp, err := abaputils.GetHTTPResponse("HEAD", connection, nil, client)
+	if err != nil {
+		err = abaputils.HandleHTTPError(resp, err, "Authentication on the ABAP system failed", con)
+		return err
+	}
+	defer resp.Body.Close()
+
+	log.Entry().WithField("StatusCode", resp.Status).WithField("ABAP Endpoint", connection.URL).Debug("Authentication on the ABAP system successful")
+	connection.XCsrfToken = resp.Header.Get("X-Csrf-Token")
+
+	errorOccurred := false
+	for _, item := range backlog {
+		err = createTagsForSingleItem(item, telemetryData, connection, client, com)
+		if err != nil {
+			errorOccurred = true
+		}
+	}
+
+	if errorOccurred {
+		message := "At least one tag has not been created"
+		log.Entry().Errorf(message)
+		return errors.New(message)
+	}
+	return nil
+
+}
+
+func createTagsForSingleItem(item CreateTagBacklog, telemetryData *telemetry.CustomData, con abaputils.ConnectionDetailsHTTP, client piperhttp.Sender, com abaputils.Communication) (err error) {
+
+	for index := range item.tags {
+		err = createSingleTag(item, index, telemetryData, con, client, com)
+	}
+	return err
```

Comment:
my golang is not that good but does this not ignore the first error in case 2 tags are created and the 2nd one worked fine there is no error at all?

Reply:
yes, i already made adaptions in another loop, but forgot this one.